packages/jsii/lib/compiler.ts

```diff
@@ -126,7 +168,10 @@ export class Compiler implements Emitter {
         ...COMPILER_OPTIONS,
         noEmitOnError: false,
       },
-      { ...ts.sys, getCurrentDirectory() { return projectRoot; } }
+      { ...ts.sys, getCurrentDirectory() { return projectRoot; } },
+      ts.createSemanticDiagnosticsBuilderProgram,
+      undefined,
+      diag => this.eventEmitter.emit('statusChanged', diag)
```

Comment:
Why does this need to go through an event emitter?

Reply:
This was to preserve the console output behavior (while avoiding the standard behavior to clear out the console when testing, which is SUPER inconvenient), but I believe I have a better way...